includedBuild/gradleConfiguration/src/main/kotlin/com/badoo/reaktive/publish/PublishConfigurationPlugin.kt

```diff
@@ -3,37 +3,102 @@ package com.badoo.reaktive.publish
 import com.badoo.reaktive.configuration.JsPlugin
 import com.badoo.reaktive.configuration.Target
-import com.jfrog.bintray.gradle.BintrayExtension
-import com.jfrog.bintray.gradle.BintrayPlugin
-import com.jfrog.bintray.gradle.tasks.BintrayUploadTask
 import org.gradle.api.Plugin
 import org.gradle.api.Project
 import org.gradle.api.publish.PublishingExtension
 import org.gradle.api.publish.maven.MavenPublication
-import org.gradle.api.publish.maven.internal.artifact.FileBasedMavenArtifact
+import org.gradle.api.publish.maven.plugins.MavenPublishPlugin
 import org.gradle.api.publish.maven.tasks.AbstractPublishToMaven
-import org.gradle.kotlin.dsl.apply
-import org.gradle.kotlin.dsl.getByType
-import org.gradle.kotlin.dsl.named
+import org.gradle.api.publish.maven.tasks.PublishToMavenLocal
+import org.gradle.api.publish.maven.tasks.PublishToMavenRepository
+import org.gradle.api.publish.plugins.PublishingPlugin
 import org.gradle.kotlin.dsl.withType
+import org.gradle.plugins.signing.SigningExtension
+import org.gradle.plugins.signing.SigningPlugin
 import org.jetbrains.kotlin.gradle.plugin.mpp.KotlinMultiplatformPlugin
+import java.net.URI
 
 class PublishConfigurationPlugin : Plugin<Project> {
 
     override fun apply(target: Project) {
-        target.apply(plugin = "maven-publish")
+        target.plugins.apply(MavenPublishPlugin::class.java)
         val taskConfigurationMap = createConfigurationMap(target)
-        createFilteredPublishToMavenLocalTask(target)
-        setupLocalPublishing(target, taskConfigurationMap)
-        setupBintrayPublishingInformation(target)
-        setupBintrayPublishing(target, taskConfigurationMap)
+        disablePublishingTasks(target, taskConfigurationMap)
+        createFilteredPublishTasks(target)
+        setupPublishing(target)
+        setupSign(target)
     }
 
-    private fun createFilteredPublishToMavenLocalTask(target: Project) {
-        // Create empty task with enabled publish tasks only
-        // We can't use regular publishToMavenLocal
-        target.tasks.register(TASK_FILTERED_PUBLISH_TO_MAVEN_LOCAL) {
-            dependsOn(project.tasks.matching { it is AbstractPublishToMaven && it.enabled })
+    private fun setupPublishing(project: Project) {
+        val publishing = project.extensions.getByType(PublishingExtension::class.java)
+        publishing.repositories {
+            maven {
+                name = "sonatype"
+                url = URI.create("https://oss.sonatype.org/service/local/staging/deploy/maven2/")
+                credentials {
+                    username = project.property("sonatype.username").toString()
+                    password = project.property("sonatype.password").toString()
+                }
+            }
+        }
+        publishing.publications.withType(MavenPublication::class.java).configureEach {
+            pom {
+                name.set(project.name)
+                description.set(project.description)
```

Comment:
Project descriptions are empty now. Not sure if it is mandatory to setup description for every module.